libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
Hmm.  It's a little bit useless -- it basically just doubles the grace period.  Maybe we can do without it.  The only case I can think of where it would help is if the Put call completely hangs on the server due to some bug, and doesn't return quickly even though the Put succeeded.  Unless you can think of a more realistic bug this would fix (that wouldn't also be fixed by just increasing the grace period), we can probably drop that logic.